added saucelabs configs

AdrianCozma opened the pull request:
this is still a work in progress

TO DO: 
- [ ] send test status to SauceLabs

Commit:
Add information in debugging in k3po.js, added some extra logging in the browser for when debugging was enabled

Commit:
Merge branch 'debugging' of https://github.com/dpwspoon/k3po.js into travis

Commit:
added SauceLabs resources

Commit:
updated readme

Commit:
updated travis.yml

Commit:
minor update

Commit:
minor update

Commit:
minor update

Commit:
trying with node 4

Commit:
node 0.12

Commit:
updated readme with saucelabs badges

Commit:
minor readme update

dpwspoon commented on .travis.yml:
```diff
@@ -1,11 +1,25 @@
+  global:
+    - NODE_VERSION=0.12
+    - SAUCE_USERNAME="kaazing-build"
+    - SAUCE_ACCESS_KEY="f350bbbb-376e-4c0e-b7b6-ae78f58f67f2"
```
How does this sauce lab key work.  Is it specific per each repo or generic for all kaazing.  Good either way, just want to understand

AdrianCozma replied:
the key is generic to the sauce labs account(kaazing-build) and it can be regenerated if needed

dpwspoon commented on Gruntfile.js:
```diff
@@ -69,19 +69,46 @@ module.exports = function (grunt) {
+                    reporter: 'spec',
+                    ui: 'mocha-k3po',
+                    require: 'lib/testFrameworks/mocha-k3po.js',
+                    captureFile: "build/testMochaK3po.txt",
```
what is this capture file.txt used for (maybe add a comment)

AdrianCozma replied:
from what I read it captures all the output of from the test run(I left it there because it was in the task to run tests locally, on which I based the sauceTest task)

Commit:
added more browsers

Commit:
used higher node version

Commit:
install compile dep for higher node version

Commit:
made sauce connect available to pull requests

AdrianCozma commented:
I added a script to install sauce connect as it seems that it is not available to builds from PRs(there is an open issue about this https://github.com/travis-ci/travis-ci/issues/2135), I also modified the node.js version to `stable`(we were using 0.12 and the latest stable version is 5; do we need run on any specific node.js version?)

I've added more browsers and some are indeed failing as you predicted.

Also I'm getting two types on errors that are not related to the tests themselves and don't always reproduce:
- k3po start error; [here](https://travis-ci.org/k3po/k3po.js/jobs/103380753) and [here](https://travis-ci.org/k3po/k3po.js/jobs/103380760) notice that it fails at different stages but with the same error
- 2nd k3po start error; [here](https://travis-ci.org/k3po/k3po.js/jobs/103380755); this error does not trigger a failure and results in a false positive
- there is also this [build](https://travis-ci.org/k3po/k3po.js/jobs/103340305) which failed because it couldn't connect to sauce labs(this is fixed now by the sauce connect setup script) which also did not trigger a failure

I'll look a bit more into these errors tommorow but if you can provide any insight on them(or how to make grunt stop giving false positives) it would be of great help.

Thanks

dpwspoon commented on lib/browser/RemoteScriptRunner.js:
```diff
@@ -138,7 +139,11 @@ module.exports = function () {
-            return this._cmdQueue.shift();
+            var command = this._cmdQueue.shift();
+            if(this._debug) {
+                console.log("RemoteScriptRunner received command: " + command);
```
This code runs in the browser and might not work for all browsers, see: http://patik.com/blog/complete-cross-browser-console-log/.  This is okay for now because it doesn't get executed by default, but we will ant to fix this at somepoint

dpwspoon commented:
@AdrianCozma, this is all good!  We should fix the test failures.  Or ignore the failing tests in specific browsers and then add ones that will work.  IE10 should work as I believe it has websocket support and so should Firefox and Safari.  So I'm not sure what is going on there.  

The Firefox error looks very strange, like a loading problem.

The Safari error looks like the Upgrade header wasn't matched.  Maybe we can wireshark that.

IE10 it looks like something fundamental went wrong, like maybe we are using code that causes it to fail

For the other failures in browsers that do not support websocket we should ignore the tests.

I'm happy to track this and divide work on however you see fit.  If you want to file issues and then ignore the travis builds for the failing ones, I could merge this PR.  Or we could wait until its all done here.  Which ever you prefer.

dpwspoon commented:
Oops, missed your comment up above before this.  We need to get those false positives to throw actual errors.  :-(.  Not sure what the maven resolved dependencies issues are

Commit:
test commit to check job status update on Sauce Labs

Commit:
updated job status update logic on Sauce Labs

AdrianCozma commented:
@dpwspoon I added two commits to set the job result in Sauce Labs, please review them when you get the chance. 
Regarding the test failures and ignoring tests for certain browsers, I think it would be better to merge this PR(if everything is in order), and I'll open issues for them to track them better(having everything in one PR seems like it would be hard to manage)

dpwspoon commented:
@AdrianCozma +1 on the PR, I'll merge this for now as no one is using it yet.  

Nice solution with the Saucelabs integration.

dpwspoon commented:
LGTM
